### ISSUE TITLE
Feature request: -OnEnd Parameter for Enter-RSSession

### ISSUE THREAD

**Issue by Callidus2000** [mdgrs-mei/RestartableSession#2]
Hello,
It would be helpful if OnStart had an accompanying OnEnd parameter.  This would allow, for example, automatic logout from remote systems.

**Comment by mdgrs-mei** [comment]
Thank you for the suggestion. I can totally see the usecase.

I will try adding OnEnd and OnEndArgumentList. This will require renaming the current ArgumentList to OnStartArgumentList but it shouldn't be a big issue.

**Comment by Callidus2000** [comment]
Great. You could add `ArgumentList` as an alias to `OnStartArgumentList`...
Is there a possibility/way to return/keep initialized variables from the OnStart block into the session? E.g. If I initialize a DB connection within OnStart it would be cool to have access to the object in the then running RSSession.

**Comment by mdgrs-mei** [comment]
Yes. OnStart is executed in the session scope so variables defined in OnStart are all visible in the RSSession. They will be accessible from OnEnd too.

**Comment by Callidus2000** [comment]
> OnStart is executed in the session scope
Sorry, my fault. I could also have sworn that this would have worked for my first uses of the module. I moved my OnStart code to a file and then just called the script incorrectly (without the extra '.').

**Comment by mdgrs-mei** [comment]
@Callidus2000 Version 0.4.0 that includes this request has been released to PowerShell Gallery.

https://github.com/mdgrs-mei/RestartableSession/blob/main/CHANGELOG.md

I also updated the README about the scope where OnStart is executed. Please try the new feature and close this issue if it works.

**Comment by Callidus2000** [comment]
I'll try it on Monday, thanks for the fast implementation.

**Comment by Callidus2000** [comment]
Finally have given it a try, works fine.
